### DISCUSSION TITLE
What causes SeisSol's memory to keep growing during the calculation?

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#889] [Q&A]
According to my experience, the memory consumption of SeisSol is very stable for previous examples with no more than 10 clusterings, but when I calculate an example with a lot of clusterings (18), the memory consumption keeps growing, resulting in more than 1000G of memory consumed for the simulation of SeisSol with a 13 millions grid using third order basis functions with double precision, which is far more than the memory needed for the theoretical calculation, have you encountered this situation? 

![image](https://github.com/SeisSol/SeisSol/assets/104721358/da808b42-da7b-46d7-9cf6-2322986ddddb)

**Comment by Thomas-Ulrich** [comment]
that's definitely worth investigating, and I think it might relate to #691

**Reply by Thomas-Ulrich** [reply]
I've added a small region of high velocity to tpv12_13 example as follow:
```
diff --git a/tpv12_13/tpv12_13_material.yaml b/tpv12_13/tpv12_13_material.yaml
index a12c911..26f63e5 100644
--- a/tpv12_13/tpv12_13_material.yaml
+++ b/tpv12_13/tpv12_13_material.yaml
@@ -1,9 +1,21 @@
 !Switch
-[rho, mu, lambda, plastCo, bulkFriction]: !ConstantMap
+[rho, plastCo, bulkFriction]: !ConstantMap
   map:
     rho:                 2700
-    mu:           2.9403e+010
-    lambda:        2.941e+010
     plastCo:          5.0e+06
     bulkFriction:        0.85
+[mu, lambda]: !LuaMap
+  returns: [mu, lambda]
+  function: |
+      function f (x)
+        if ((math.abs(x["x"])<100.) and (math.abs(x["z"]+5000)<100.)) then
+          alpha = 200000.0
+        else
+          alpha = 1.0
+        end
+        return {
+          mu = 2.9403e10 * alpha,
+          lambda = 2.941e10 * alpha
+        }
+      end
```

, and the memory consumption indeed raises, especially when  a h5/xdmf output is written (but there are other memory leaks of lesser extend). 
Monitored with:

`watch -n 1 "sstat -j <jobid> --format="JobId,MaxRSS" -P && tail -n 10 logout.out"`

**Reply by krenzland** [reply]
Afaik the MaxRSS is the maximum over the entire simulation time (hence it cannot shrink) and is computed by sampling. When we e.g. resize a buffer, both old and new buffers exist for some time. Hence, it doesn't directly indicate a memory leak
